crates/nix_rs/src/flake/outputs.rs

```diff
@@ -1,109 +1,43 @@
-  pub static ref DEFAULT_FLAKE_SCHEMAS: FlakeUrl = {
-    Into::<FlakeUrl>::into(Path::new(env!("DEFAULT_FLAKE_SCHEMAS")))
-  };
-
-  /// Flake URL of the flake that defines functions for inspecting flake outputs
-  ///
-  /// We expect this environment to be set in Nix build and shell.
-  pub static ref INSPECT_FLAKE: FlakeUrl = {
-    Into::<FlakeUrl>::into(Path::new(env!("INSPECT_FLAKE")))
-  };
-}
-
-/// Represents the "outputs" of a flake
-///
-/// TODO: Rename this to `FlakeSchema` while generalizing the existing `schema.rs` module.
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
-pub struct FlakeOutputs {
-    pub inventory: BTreeMap<String, InventoryItem>,
-}
+use super::schema::Val;
 
+/// Flake outputs derived from [super::schema::FlakeSchemas]
 #[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
 #[serde(untagged)]
-pub enum InventoryItem {
-    Leaf(Leaf),
-    Attrset(BTreeMap<String, InventoryItem>),
+pub enum FlakeOutputs {
+    Val(Val),
+    Attrset(BTreeMap<String, FlakeOutputs>),
 }
 
 impl FlakeOutputs {
-    /// Determine flake outputs using [static@INSPECT_FLAKE] and [static@DEFAULT_FLAKE_SCHEMAS]
-    pub async fn from_nix(
-        nix_cmd: &crate::command::NixCmd,
-        flake_url: &super::url::FlakeUrl,
-        system: &super::System,
-    ) -> Result<Self, crate::command::NixCmdError> {
-        let inspect_flake: FlakeUrl = INSPECT_FLAKE
-            // Why `exculdingOutputPaths`?
-            //   This function is much faster than `includingOutputPaths` and also solves <https://github.com/juspay/omnix/discussions/231>
-            //   Also See: https://github.com/DeterminateSystems/inspect/blob/7f0275abbdc46b3487ca69e2acd932ce666a03ff/flake.nix#L139
-            //
-            //
-            // Note: We might need to use `includingOutputPaths` in the future, when replacing `devour-flake`.
-            // In which case, `om ci` and `om show` can invoke the appropriate function from `INSPECT_FLAKE`.
-            //
-            .with_attr("contents.excludingOutputPaths");
-        let systems_flake = SystemsListFlakeRef::from_known_system(system)
-            // TODO: don't use unwrap
-            .unwrap()
-            .0
-            .clone();
-        let flake_opts = FlakeOptions {
-            no_write_lock_file: true,
-            override_inputs: BTreeMap::from_iter([
-                (
-                    "flake-schemas".to_string(),
-                    DEFAULT_FLAKE_SCHEMAS.to_owned(),
-                ),
-                ("flake".to_string(), flake_url.clone()),
-                ("systems".to_string(), systems_flake),
-            ]),
-            ..Default::default()
-        };
-        let v = nix_eval::<Self>(nix_cmd, &flake_opts, &inspect_flake).await?;
-        Ok(v)
-    }
-}
-
-impl InventoryItem {
-    /// Get the non-attrset leaf
-    pub fn as_leaf(&self) -> Option<&Leaf> {
+    /// Get the non-attrset value
+    pub fn as_val(&self) -> Option<&Val> {
```

Comment:
Likewise, `get_value`?

(API naming is generally best reflective of semantics than Rust types)

Reply:
https://github.com/juspay/omnix/pull/235/commits/c877c77653fb248ece5add7ab7741c8055d3635a